src/sth_database.js

```diff
@@ -0,0 +1,181 @@
+/* globals module, process, require */
+
+(function() {
+  "use strict";
+
+  var mongoose = require('mongoose');
+
+  var sthConfig, sthHelper, connectionURL, eventSchema, aggregatedSchema;
+
+  /**
+   * Connects to a (MongoDB) database endpoint asynchronously
+   * @param {string} authentication The authentication schema to use for the connection
+   * @param {string} host The host hosting the database
+   * @param {string} port The port where the database server is listening to
+   * @param {string} database The name of the database to connect to
+   * @param {Function} callback A callback to inform about the result of the operation
+   */
+  function connect(authentication, host, port, database, callback) {
+    connectionURL = 'mongodb://' + authentication + '@' + host + ':' + port +
+    '/' + database;
+
+    mongoose.connect(connectionURL, function (err) {
+      if (err) {
+        // Error when connecting to the MongoDB database
+        return callback(err);
+      }
+
+      // Event collection schema
+      eventSchema = mongoose.Schema({
```

Comment:
Not really, the mongoose.Schema() is a function of the mongoose Node.js module (MongoDB driver) which let us declare the schema to be used. So to day, it allows the developer to define the schema of the docs on the client side. This schema can be used afterwards to compile (to instantiate) "Models" (using mongoose terminology) which will map to concrete collections in MongoDB.
So to say it, mongoose defines schemas and models "to model" the structure of the docs to be stored in a MongoDB database. But they are only used on the client side. Regarding MongoDB, it just stores docs (JSON objects) into collections using certain internal structure. In fact, MongoDB allows storing docs with distinct internal structures into the same collection.
So to say, it is the way mongoose let developers deal with docs and their internal structure.
I hope I made myself clear :)